homeassistant/components/eufylife_ble/sensor.py

```diff
@@ -0,0 +1,207 @@
+"""Support for EufyLife sensors."""
+from __future__ import annotations
+
+from contextlib import suppress
+from typing import Any
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth import async_address_present
+from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.const import ATTR_UNIT_OF_MEASUREMENT, STATE_UNAVAILABLE, UnitOfMass
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.util.unit_conversion import MassConverter
+from homeassistant.util.unit_system import US_CUSTOMARY_SYSTEM
+
+from .const import DOMAIN, EUFYLIFE_MODEL_TO_NAME
+from .models import EufyLifeData
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the EufyLife sensors."""
+    data: EufyLifeData = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [
+            EufyLifeWeightSensorEntity(data),
+            EufyLifeRealTimeWeightSensorEntity(data),
+        ]
+    )
+    if data.client.supports_heart_rate:
+        async_add_entities([EufyLifeHeartRateSensorEntity(data)])
+
```

Comment:
Please only call `async_add_entities` once since it schedules a task

Reply:
Fixed.